src/python/dxpy/scripts/dx.py

```diff
@@ -2513,22 +2513,18 @@ def install(args):
 
 def uninstall(args):
     app_desc = get_app_from_path(args.app)
-    if app_desc is None:
-        try:
-            dxpy.api.app_uninstall(args.app)
-            print('Uninstalled application with id: ' + args.app)
-        except:
-            try:
-                dxpy.api.app_uninstall('app-' + args.app)
-                print('Uninstalled application with name: ' + args.app)
-            except:
-                err_exit()
+    if app_desc:
+        try_call(dxpy.api.app_uninstall, app_desc['id'])
+        print('Uninstalled the {name} app'.format(name=app_desc['name']))
     else:
-        try:
-            dxpy.api.app_uninstall(app_desc['id'])
-            print('Uninstalled the ' + app_desc['name'] + ' app')
-        except:
-            err_exit()
+        user_data = dxpy.api.user_describe(dxpy.whoami(), {"fields": {"appsInstalled": True}})
+        if args.app in user_data['appsInstalled']:
+            try_call(dxpy.api.app_uninstall, 'app-' + args.app)
+        elif args.app.startwith('app-') and args.app[len('app-'):] in user_data['appsInstalled']:
+            parser.exit(1, 'Could not find the app\n')
```

Comment:
Not exactly, line 2526 we using when user trying to uninstall with app ID. If we continue support uninstalling by ID, we should accept any values, not only from 'user_describe' list. Additionally we can check 'app-' prefix or parameter length to find out if this is ID, or accept any and parse 404 response for custom 'app not found' message, but if this make sense?